tests/framework/e2e/cluster.go

```diff
@@ -780,7 +781,18 @@ func (epc *EtcdProcessCluster) CloseProc(ctx context.Context, finder func(EtcdPr
 	return proc.Close()
 }
 
-func (epc *EtcdProcessCluster) StartNewProc(ctx context.Context, cfg *EtcdProcessClusterConfig, tb testing.TB, opts ...config.ClientOption) error {
+type MemberAddFunc func(ctx context.Context, ctlv3 *EtcdctlV3, name string, peerAddrs []string) (*clientv3.MemberAddResponse, error)
+
+var (
+	MemberAdd = func(ctx context.Context, ctlv3 *EtcdctlV3, name string, peerAddrs []string) (*clientv3.MemberAddResponse, error) {
+		return ctlv3.MemberAdd(ctx, name, peerAddrs)
+	}
+	MemberAddAsLearner = func(ctx context.Context, ctlv3 *EtcdctlV3, name string, peerAddrs []string) (*clientv3.MemberAddResponse, error) {
+		return ctlv3.MemberAddAsLearner(ctx, name, peerAddrs)
+	}
+)
+
+func (epc *EtcdProcessCluster) StartNewProc(ctx context.Context, cfg *EtcdProcessClusterConfig, tb testing.TB, memberAddFunc MemberAddFunc, opts ...config.ClientOption) (memberID uint64, err error) {
```

Comment:
Can we just add a parameter something like `addAsLearner bool` for `StartNewProc`? It's much simpler and looks more natural.